ts/Extensions/Boost/BoostSeries.ts

```diff
@@ -849,6 +845,76 @@ function getPoint(
     return point;
 }
 
+/**
+ * @private
+ */
+function scatterProcessData(
+    series: Series
+): (boolean|undefined) {
+
+    if (series.isDirtyData) {
+        return;
+    }
+
+    if (!series.boost) {
+        series.data.length = 0;
+        series.isDirty = true;
+        return;
+    }
+
+    series.yAxis.setTickInterval();
+
+    const {
+            max: xMax,
+            min: xMin
+        } = series.xAxis.getExtremes(),
+        {
+            max: yMax,
+            min: yMin
+        } = series.yAxis.getExtremes();
+
+    if (
+        typeof yMin !== 'number' ||
+        typeof yMax !== 'number'
+    ) {
+        return;
+    }
+
+    const processedData: Array<PointOptions> = [],
+        processedXData: Array<number> = [],
+        processedYData: Array<number> = [],
+        xData = series.xData || [],
+        yData = series.yData || [];
+
+    let cropped = false,
+        x: number,
+        y: number;
+
+    for (let i = 0, iEnd = xData.length; i < iEnd; ++i) {
+        x = xData[i];
+        y = yData[i] as number;
+
+        if (
+            x >= xMin && x <= xMax &&
+            y >= yMin && y <= yMax
+        ) {
+            processedData.push({ x, y });
+            processedXData.push(x);
+            processedYData.push(y);
+        } else {
+            cropped = true;
+        }
+    }
+
+    series.cropped = cropped;
+    series.cropStart = 0;
+    series.processedData = processedData;
+    series.processedXData = processedXData;
+    series.processedYData = processedYData;
```

Comment:
Do we need all these properties? As far as I can see, cropping don't make sense in this scenario.

Also, we shouldn't have to keep both `processedXData`/`processedYData` and `processedData`. In fact, the code in `seriesRenderCanvas` shows that `processedData` is not used whenever `processedXData` is around (through the statement `useRaw = !xData`). I suggest we remove `processedData` and focus on efforts on `proxessedXData` and `processedYData`.

Reply:
This is for `Series.generatePoints` to pick up the right amount points after the boost exit. So this covering both situations: `enterBoost` with `Series.renderCanvas` and `exitBoost` with `Series.render`.